### PR TITLE
Upgrade minimum supported version to 21.8 in ClickHouse 

### LINKED ISSUE #14112: Upgrade minimum supported version in ClickHouse connector
EOL of `21.3.2.5` is 30 Mar 2022

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Upgrade minimum supported version to 21.8 in ClickHouse 
Fixes #14112

## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# ClickHouse
* Upgrade minimum supported version to 21.8. ({issue}`14112`)
```

### CODE REVIEW

**Review by hashhar — COMMENTED** [review]
I assume the motivation is to cleanup some code since we move from one EOL version to another EOL version?

**Comment by tangjiangling** [comment]
> I assume the motivation is to cleanup some code since we move from one EOL version to another EOL version?

IIUC the main purpose is to support escaping comment values, however unfortunately older versions of ClickHouse do not support table comment. (see https://github.com/trinodb/trino/pull/14088#discussion_r967625487)

**Comment by ebyhr** [comment]
> I assume the motivation is to cleanup some code

Correct. We could remove support for 21.8 too, but I feel it's too early.

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseTypeMapping.java [hunk 1]
```diff
@@ -27,11 +25,6 @@ protected QueryRunner createQueryRunner()
             throws Exception
     {
         clickhouseServer = closeAfterClass(new TestingClickHouseServer());
-        return createClickHouseQueryRunner(clickhouseServer, ImmutableMap.of(),
-                ImmutableMap.<String, String>builder()
-                        .put("metadata.cache-ttl", "10m")
-                        .put("metadata.cache-missing", "true")
-                        .buildOrThrow(),
-                ImmutableList.of());
+        return createClickHouseQueryRunner(clickhouseServer);
```
Comment (hashhar) [inline plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseTypeMapping.java:28]: any specific reason to remove metadata cache from type mapping tests?

**Review by hashhar — APPROVED** [review]

**Review by tangjiangling — APPROVED** [review]

File: plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseTypeMapping.java (hunk 1 above)
Comment (ebyhr) [reply plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseTypeMapping.java:28]: Here's elapsed time comparison in my laptop. No improvement by metadata cache.
* with cache: 1m 36s
* without cache: 1m 33s

**Pull request merged**